### PR TITLE
Rename with same name different case should not fail

### PR DESCRIPTION
**Pull request by AdityaHegde**
- [x] Rename file with same name different case should not fail.
- [x] Rename in reconcile with same different case should not fail.
- [x] DAG in reconcile should be case insensitive

### CODE REVIEW

File: runtime/drivers/file/repo.go [hunk 1]
```diff
@@ -113,14 +114,32 @@ func (c *connection) Put(ctx context.Context, instID string, filePath string, re
...
+			return err
+		}
+	} else {
+		if _, err := os.Stat(filePath); err == nil {
+			return drivers.ErrFileAlreadyExists
+		}
+		err := os.Rename(from, filePath)
+		if err != nil {
+			return err
+		}
```
Comment (begelundmuller) [inline runtime/drivers/file/repo.go:142]: Why doesn't `os.Rename` work for renaming with different case?

Feels dangerous to delete the file, even if just temporarily

File: runtime/services/catalog/migrations.go [hunk 2]
```diff
@@ -20,6 +21,7 @@ import (
 
 type MigrationItem struct {
 	Name           string
+	LowerName      string
```
Comment (begelundmuller) [inline runtime/services/catalog/migrations.go:24]: Maybe `NormalizedName` instead?

File: runtime/services/catalog/migrator/migrator.go [hunk 3]
```diff
@@ -80,7 +81,12 @@ func GetDependencies(ctx context.Context, olap drivers.OLAPStore, catalog *drive
 		// no error here. not all migrators are needed
 		return []string{}
 	}
-	return migrator.GetDependencies(ctx, olap, catalog)
+	// convert dependencies to lower case here
+	dependencies := migrator.GetDependencies(ctx, olap, catalog)
+	for i, dep := range dependencies {
+		dependencies[i] = strings.ToLower(dep)
+	}
+	return dependencies
```
Comment (begelundmuller) [inline runtime/services/catalog/migrator/migrator.go:89]: Maybe safer to do the lowercase/normalization in the migration logic and then continue returning the outright values here? To make it easier to reason about when names are normalized and when they are not

**Review by begelundmuller — CHANGES_REQUESTED** [review]

File: runtime/drivers/file/repo.go (hunk 1 above)
Comment (AdityaHegde) [reply runtime/drivers/file/repo.go:142]: You are right. This is not necessary. Not sure what lead me to believe this is needed

File: runtime/drivers/sqlite/catalog.go [hunk 4]
```diff
@@ -131,6 +132,6 @@ func (c *connection) UpdateEntry(ctx context.Context, instanceID string, e *driv
 }
 
 func (c *connection) DeleteEntry(ctx context.Context, instanceID string, name string) error {
-	_, err := c.db.ExecContext(ctx, "DELETE FROM catalog WHERE instance_id = ? AND name = ?", instanceID, name)
+	_, err := c.db.ExecContext(ctx, "DELETE FROM catalog WHERE instance_id = ? AND LOWER(name) = ?", instanceID, strings.ToLower(name))
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply runtime/drivers/sqlite/catalog.go:135]: Makes sense. Will add Lower around `?` instead.

File: runtime/drivers/file/repo.go [hunk 5]
```diff
@@ -113,14 +114,32 @@ func (c *connection) Put(ctx context.Context, instID string, filePath string, re
 // Rename implements drivers.RepoStore
 func (c *connection) Rename(ctx context.Context, instID string, from string, filePath string) error {
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply runtime/drivers/file/repo.go:115]: Ya just filePath does feel a little out of place.

**Pull request merged**